lib/options.nix

```diff
@@ -187,6 +189,188 @@ rec {
   */
   mkPackageOptionMD = mkPackageOption;
 
+  /* Creates a suggestion based resource namespace allocator.
+
+    It allows someone to define a value space, like a port interval, so other
+    modules can "ask" for a value and the module will check for missing values,
+    conflicts and invalid candidates and suggest values for these cases that
+    must be explicitly set.
+
+    See RFC 0159 for more specification details.
+  */
+  mkAllocModule = {
+    # ---- begin of passthru parameters to the outer mkOption
+    example ? null,
```

Comment:
Documentation needed.

Reply:
Documentation added

I also used XML comments in those section comments to guide the ones reading code without messing up with the generated documentation